### PR TITLE
Use codecov

### PR DESCRIPTION
**Pull request by ntwb**
> Which issue, if any, is this issue related to?

#2880

> Is there anything in the PR that needs further explanation?

@gucong3000 This is a PR for the branch you created

Please in the future make sure each branch has a pull request associated with it

### CODE REVIEW

**Comment by ntwb** [comment]
Some notes:

• Please do not make any changes to Travis CI in this PR, for testing per the original discussion only add codecov to AppVeyor for now please.

• There are also significant changes and unnecessary changes being made to the `package.json` file, the only change that should be required is the addition of `"codecov": "^2.3.0",` I think.

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ matrix:
       script: npm run pretest
       env: CI=pretest
     - node_js: '8'
-      script: npm run jest -- --runInBand  --coverage
+      script: npm run jest -- --runInBand --coverage
```
Comment (ntwb) [inline .travis.yml]: This change should also be removed, it should be its own pull request please

Note: _Pull requests should only ever contain the relevant changes required for the intended changes of the pull request._

**Comment by jeddy3** [comment]
Is codecov compatible with travis? Could we try it there as it seems appveyor is running out of memoery.

**Comment by gucong3000** [comment]
> Is codecov compatible with travis? 

Yes.

> Could we try it there as it seems appveyor is running out of memoery.

I've tried, and travis working normally.
https://codecov.io/gh/stylelint/stylelint/branch/codecov-test
https://codecov.io/gh/stylelint/stylelint/commit/67cdcec5b8d59c8d9d059e138688d4be46a95d39/build

**Comment by ntwb** [comment]
Thanks for trying this @gucong3000 😄 

I'm going to close this per the comment I made here https://github.com/stylelint/stylelint/issues/2880#issuecomment-331337592
